### DISCUSSION TITLE
Is there any API for Discussions in GitHub?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#23483] [New to GitHub]
<p>Hi all!</p>
<h4>
<a name="h-1-context-1" class="anchor" href="#h-1-context-1"></a>1. Context</h4>
<p>I would like to know where I can find API information from Discussions. For example, Gitlab has an API for you to receive and send information from Discussions. Is there any API for Discussions in GitHub?</p>
<h4>
<a name="h-2-notes-2" class="anchor" href="#h-2-notes-2"></a>2. Notes</h4>
<ol>
<li>I’m not criticizing Github, just asking if there is such a possibility, the possibility of having an API for the Discussions area</li>
<li>I didn’t find information about my question or the documentation links about the API from GitHub Discussions - that’s why i’m asking here</li>
<li>By mentioning the Gitlab Discussions API, I don’t want to promote Gitlab, I just reference the Gitlab resource on GitHub, with this aim to check if there is such a possibility of a resource on GitHub - resource: the possibility of having an API for the Discussions area</li>
<li>I would be happy if someone could clarify this doubt if possible</li>
</ol>
<h4>
<a name="h-3-references-3" class="anchor" href="#h-3-references-3"></a>3. References</h4>
<ul>
<li><a href="https://docs.gitlab.com/ee/api/discussions.html" class="inline-onebox" rel="noopener nofollow ugc">Discussions API | GitLab</a></li>
</ul>

**Comment by mpboom** [comment, marked as answer]
<p>Hi <a class="mention" href="/nkot56297">@nkot56297</a> ! There is, for GraphQL:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions">
  <header class="source">
      <img src="https://docs.github.com/assets/cb-803/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/19948032/181077341-703d0410-e83f-45fe-9217-08c17196a4a6.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions" target="_blank" rel="noopener nofollow ugc">Using the GraphQL API for Discussions - GitHub Docs</a></h3>

  <p>Learn how to use the GitHub Discussions GraphQL API.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by ettaboyle** [comment]
<p>Hey <a class="mention" href="/nkot56297">@nkot56297</a> :wave:t2: I’m gonna +1 <a class="mention" href="/mpboom">@mpboom</a>’s comment and also :eyes: tag this for the PM on the Discussions team <a class="mention" href="/evi-liu">@evi-liu</a> as a feature request.</p>
<p>If you have <a href="https://github.com/github/feedback/discussions/categories/discussions-feedback">feature requests or bugs on Discussions</a> (or most of our other products) you can share them in the <a href="https://github.com/github/feedback/discussions">feedback/discussion</a> – thank yu :sparkles:</p>

**Reply by CodingwithFawad** [reply]
Thank you!! :

**Comment by ghost** [comment]
<p><a class="mention" href="/ettaboyle">@ettaboyle</a> <a class="mention" href="/mpboom">@mpboom</a> Hi all!</p>
<ol>
<li>thank you for feedback</li>
<li>I have a lot of ideas to make GitHub an awesome place please someone see these ideas:<br>
<a href="https://github.com/github/feedback/discussions?discussions_q=author:nkot56297" rel="noopener nofollow ugc">GitHub - Discussions Feedback - discussions?discussions_q=author:nkot56297</a>
</li>
<li>Here my profile in GitHub: <a href="https://github.com/nkot56297" class="inline-onebox" rel="noopener nofollow ugc">nkot56297 (nkot56297) · GitHub</a>
</li>
</ol>

**Reply by CodingwithFawad** [reply]
Thank you!! :

**Comment by ettaboyle** [comment]
<p>Thank you!! :slight_smile: We always appreciate the feedback &amp; suggestions</p>

**Comment by xamidi** [comment]
Well, the correct answer is: Yes, there is.

For example: [`https://api.github.com/repos/xamidi/pmGenerator/discussions/2`](https://api.github.com/repos/xamidi/pmGenerator/discussions/2)

But it is currently not mentioned in the repo's JSON (such as [`https://api.github.com/repos/xamidi/pmGenerator`](https://api.github.com/repos/xamidi/pmGenerator)).

Also the `/reactions` and `/timeline` subrequests seem to be not working, despite being listed.

But there is a working undocumented [`/comments`](https://api.github.com/repos/xamidi/pmGenerator/discussions/2/comments) subrequest.

It seems to be working similar to [Team Discussions](https://docs.github.com/en/rest/teams/discussion-comments?apiVersion=2022-11-28#list-discussion-comments).
